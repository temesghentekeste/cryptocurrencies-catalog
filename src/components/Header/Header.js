import { useState, useEffect } from 'react';
import { Link, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

import styles from './Header.module.css';

const Header = ({ handleFilter }) => {
  const [keyword, setKeyword] = useState('');
  const history = useHistory();

  useEffect(() => {
    handleFilter(keyword);
    history.push('/');
  }, [keyword]);

  const handleChange = (e) => {
    setKeyword(e.target.value);
  };

  const handleClick = (e) => {
    e.preventDefault();
    handleFilter(keyword);
    history.push('/');
  };

  return (
    <header className={styles.header} data-testid="header">
      <div className={styles.header__right}>
        <Link to="/">
          <h1>Cryptocurrencies</h1>
        </Link>

        <Link to="/trending">
          <h3>Trending Dashboard</h3>
        </Link>
      </div>
      <div className={styles.header__search}>
        <input
          type="text"
          name="keyword"
          placeholder="Search for something. Ex. iExec, Bitcoin, DOGEUSD..."
          value={keyword}
          onChange={(e) => handleChange(e)}
        />
        <form onSubmit={handleClick}>
          <input type="submit" value="Search" className={styles.btnSubmit} />
        </form>
      </div>
    </header>
  );
};

Header.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Header;
