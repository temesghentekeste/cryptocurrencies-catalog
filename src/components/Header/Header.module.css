.header {
  margin: 2rem 0;
  padding: 1rem;
  background: linear-gradient(
    90deg,
    rgb(109, 111, 112) 0%,
    rgba(130, 120, 143, 0.897) 25%
  );
  color: #fff;

  display: flex;
  justify-content: space-between;
}

.header h1 {
  font-size: 1.8rem;
}

.header input {
  flex: 0 0 45%;
  padding: 0.4rem 0.8rem;
  outline: none;
}
