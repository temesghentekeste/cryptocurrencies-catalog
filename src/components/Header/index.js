import { useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import styles from './Header.module.css';

const Header = ({ handleFilter }) => {
  const [keyword, setKeyword] = useState('');

  const handleChange = (e) => {
    setKeyword(e.target.value);
    handleFilter(keyword);
  };

  const handleClick = () => {
    handleFilter(keyword);
  };

  return (
    <header className={styles.header}>
      <Link to="/">
        <h1>Cryptocurrencies</h1>
      </Link>
      <div className={styles.header__search}>
        <input
          type="text"
          placeholder="Search for something. Ex. iExec, Bitcoin, DOGEUSD..."
          value={keyword}
          onChange={(e) => handleChange(e)}
        />
        <button type="button" onClick={handleClick}>
          Search
        </button>
      </div>
    </header>
  );
};

Header.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Header;
