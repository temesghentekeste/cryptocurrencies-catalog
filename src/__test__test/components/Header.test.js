import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '../../components/Header/';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Router>
      <Header
        handleFilter={() => {
          console.log('testing...');
        }}
      />
    </Router>
  );
  getByTestId = component.getByTestId;
});

it('renders the Header component', () => {
  const component = renderer
    .create(
      <Router>
        <Header
          handleFilter={() => {
            console.log('testing...');
          }}
        />
      </Router>
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});
