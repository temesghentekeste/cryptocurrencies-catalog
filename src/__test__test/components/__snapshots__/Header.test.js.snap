// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header component 1`] = `
<header
  className="header"
  data-testid="header"
>
  <div
    className="header__right"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Cryptocurrencies
      </h1>
    </a>
    <a
      href="/trending"
      onClick={[Function]}
    >
      <h3>
        Trending Dashboard
      </h3>
    </a>
  </div>
  <div
    className="header__search"
  >
    <input
      name="keyword"
      onChange={[Function]}
      placeholder="Search for something. Ex. iExec, Bitcoin, DOGEUSD..."
      type="text"
      value=""
    />
    <form
      onSubmit={[Function]}
    >
      <input
        className="btnSubmit"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</header>
`;
